wayland: stop panel "dodging" mouse and moving to top or left

lukefromdc opened the pull request:
Disable grabbing and moving expanded panels with the mouse in wayland due to tendency of panel other than on top to jump to top or left

lukefromdc commented:
Fixes
https://github.com/mate-desktop/mate-panel/issues/1156
and
https://github.com/mate-desktop/mate-panel/issues/1477

Replacement for https://github.com/mate-desktop/mate-panel/pull/1391 which cannot be reopened by restoring the PR branch which had deleted here but which I had kept locally. Rebased against current master of course

lukefromdc commented:
Latest force-push added a FIXME to this code as suggested in 2023 by @raveit65 , does not build this code unless wayland is enabled at build time, and looks for running under wayland rather than for not running under x11 for compatability with
https://github.com/mate-desktop/mate-panel/pull/1479
in case we want to do that

lukefromdc commented:
If anyone objects to this I can make it optional by a gsettings value or (with more work) a menuitem in the panel properties dialog

mark-herbert42 commented:
Tested on wayland - works fine for me. Panel does not ocasionally got moved to top or left and stays exactly where it is specified by properties dialog.

lukefromdc commented:
Just rebased against master, stopped the cursor from changing to a hand when the panel cannot be grabbed, and updated the inline comments to reflect that we cannot drag a nonexpanded wayland panel along a screen edge as we have its position hardcoded to the center using gtk-layer-shell.

FIXME's included to mark the spots we have grabbing the panel disabled in case anyone ever figures out how to make dragging the panel work in wayland without it jumping back to the top or left.

lukefromdc commented:
@raveit65 I believe your original issues with https://github.com/mate-desktop/mate-panel/pull/1391 have been fixed here

lukefromdc commented:
Merging, this affects only wayland and I've been using it a month with good results
